src/Compilers/Server/VBCSCompilerTests/DesktopBuildClientTests.cs

```diff
@@ -137,6 +144,19 @@ public void ConnectToServerFails()
             }
 
             [Fact]
+            public void ConnectToPipeWithSpinWait()
+            {
+                // No server should be started with the current pipe name
+                var client = CreateClient();
+                var oneSecondInMs = (int)TimeSpan.FromSeconds(1).TotalMilliseconds;
+                Assert.False(client.TryConnectToNamedPipeWithSpinWait(oneSecondInMs));
```

Comment:
You are doing a spin wait in debug mode only, thus your behavior in debug mode and in production mode will differ. Having side-effects like this in assertions seems like a bad idea.